src/graphics/graphics.js

```diff
@@ -1014,6 +1014,38 @@
          */
         TEXTURELOCK_WRITE: 2,
 
+        /**
+         * @constant
+         * @name pc.TEXTURETYPE_GENERAL
+         * @type {number}
+         * @description Texture is a general type.
+         */
+        TEXTURETYPE_GENERAL: 1,
```

Comment:
maybe should be texture semantic instead of type?

Reply:
I prefer type over semantic, which sounds to me like a vertex thing. I'm happy to change if there is consensus though, @willeastcott @raytranuk what do you think?

Reply:
`pc.TEXTURETYPE_RGB`? 😆  I've got more of a problem with `GENERAL` than `TYPE`. But I prefer `TYPE` to `SEMANTIC`.

Reply:
Also, what about `SRGB` textures? Does this fall under this enum? Or a separate flag?

Reply:
Agreed on the general naming, it bugged me too :) Though RGB is not entirely accurate either, could be RGBA.

Linear vs gamma is a little subtle. Usually it is inferred from the texture usage (normal maps are assumed 'linear' and diffuse are assumed srgb), but also the texture's pixel format (floating point textures are stored in linear space and everything else is in gamma space).

Reply:
Maybe we should add `TEXTURETYPE_RGB` and `TEXTURETYPE_RGBA` then. :)

Reply:
I don't know of a need to differentiate between those two...

Reply:
That's better than `GENERAL`.... 😄